### DISCUSSION TITLE
Image Byte Array Converter - Convert images to data files for easy image rendering

### DISCUSSION THREAD

**Discussion by jamsinclair** [witnessmenow/ESP32-Cheap-Yellow-Display#249] [Show and tell]
I am not sure if anyone else had this issue? I struggled initially trying to output my own small images onto my CYD.

After stumbling through documentation and code examples I observed some images were kept as raw RGB data in a cpp file and this was a "Eureka!" moment which made my hobby project much easier to deal with.

I initially built a tiny simple tool to create these files for myself. I decided I may as well prettify it up and share with the rest of the community 🤓

**Website**: https://image2bytearray.netlify.app/
**Repository**: https://github.com/jamsinclair/image2bytearray

After spending too much time on this, I did discover there's a much better tool at https://javl.github.io/image2cpp/ 😅 . It's great if you have more complex image requirements or a different display that you need to generate image data files for.

Maybe some of you may find it useful too!

**Comment by clach04** [comment]
Adding to the list but for color (and grayscale/greyscale):

  * https://github.com/rdagger/micropython-ili9341/blob/master/utils/img2rgb565.py does what the name says 😉 
  * https://github.com/peterhinch/micropython-nano-gui/blob/master/img_cvt.py can generate 4-bit, 8-bit, and 4-bit grayscale BUT source needs to be ppm/pgm
  * https://github.com/clach04/cyd_clocks/blob/main/image_converter.py - (as of 2025-02-09) This ONLY outputs 4-bit binary files for use with [nano-gui](https://github.com/peterhinch/micropython-nano-gui) (etc.)
